torch_xla/core/dynamo_bridge.py

```diff
@@ -280,7 +280,10 @@ def extract_compiled_graph(xla_model: torch.fx.GraphModule, xla_args):
   torch_xla._XLAC._clear_pending_irs(str(xm.xla_device()))
 
   def optimized_mod(*args):
-    torch_xla._XLAC._xla_sync_multi(args, [])
+    # mark_step needs to be blocking since we want to access args's XLADatas
+    # and they can't be placeholder.
+    if any(torch_xla._XLAC._check_tensor_need_materialization(args)):
+      xm.mark_step(wait=True)
```

Comment:
In general, does a synchronous `mark_step` can be slower than `_xla_sync_multi`? Since the latter does not wait for the execution.

Reply:
`_xla_sync_multi` is also synchronous, but it might be executing a smaller graph.

Reply:
Ah, I didn't realize that the default behavior of 'mark_step (not wait for compute)' and '_xla_sync_multi (wait for compute)' are different. Thanks for clarification!